codestodians/food-basket-api
# [Bg: #154309798]  Add token auth to request

#### What does this PR do?
- Adds an authentication middleware that validates users

#### Description of Task to be completed?
- Add auth middleware 
- Add middleware setting to default setting 

#### How should this be manually tested?
- Check out into the ` bg-add-permission-user-154309798` branch.
- Run update user mutation with and without a token

#### Any background context you want to provide?
- Previously,updating a user detail takes place with or without a token. Henceforth, user token must be added to the header of the request
 
#### Additional Info/ Questions that may arise
- In the auth middleware, an error is thrown only if the operation is a mutation and it performs an update. All queries as at now do not require a token(This could be updated as required)
- The middleware created has to be passed directly to the test for it to be used by the test.
- `HttpRequest` allows us to use a `create-a-request` object that can be accessed with dot notation.
- In the user test, the token is deleted before the snaphot is compared. This is because different tokens are generated each time the test runs and as such, deleting the token was the option(I would welcome a different option)
- To test this add the token to the header with key `AUTHORIZATION`. See screenshot below.

#### What are the relevant pivotal tracker stories?
- [#154309798](https://www.pivotaltracker.com/story/show/154309798)

#### Screenshot
<img width="938" alt="screen shot 2018-01-25 at 3 09 02 pm" src="https://user-images.githubusercontent.com/20999363/35392343-d37a9ae0-01e1-11e8-954a-a16b99d3c3da.png">
